src/main/java/games/strategy/triplea/settings/ClientSettingUiBinding.java

```diff
@@ -0,0 +1,213 @@
+      SettingType.GAME,
+      ClientSetting.SHOW_BATTLES_WHEN_OBSERVING,
+      "Whether to show a battle if you are only observing."),
+  SHOW_BETA_FEATURES_BINDING(
+      "Show Beta Features",
+      SettingType.TESTING,
+      ClientSetting.SHOW_BETA_FEATURES,
+      "Toggles whether to show 'beta' features. These are game features that are still "
+          + "under development and potentially may not be working yet.\n"
+          + "Restart to fully activate"),
+  MAP_LIST_OVERRIDE_BINDING(
+      "Map List Override",
+      SettingType.TESTING,
+      SelectionComponentFactory.filePath(ClientSetting.MAP_LIST_OVERRIDE),
+      "Overrides the map listing file specified in 'game_engine.properties'. You can for example download a copy of the"
+          + "listing file, update it, and put the path to that file here."),
+  TEST_LOBBY_HOST_BINDING(
+      "Lobby Host Override",
+      SettingType.TESTING,
+      SelectionComponentFactory.textField(ClientSetting.TEST_LOBBY_HOST),
+      "Overrides the IP address or hostname used to connect to the lobby. Useful for connecting to a test lobby."),
+  TEST_LOBBY_PORT_BINDING(
+      "Lobby Port Override",
+      SettingType.TESTING,
+      SelectionComponentFactory.intValueRange(ClientSetting.TEST_LOBBY_PORT, 1, 99999),
+      "Specifies the port for connecting to a test lobby."),
+  TRIPLEA_FIRST_TIME_THIS_VERSION_PROPERTY_BINDING(
+      "Show First Time Prompts",
+      SettingType.GAME,
+      ClientSetting.TRIPLEA_FIRST_TIME_THIS_VERSION_PROPERTY,
+      "Setting to true will trigger for any first time prompts to be shown"),
+  SAVE_GAMES_FOLDER_PATH_BINDING(
+      "Saved Games Folder",
+      SettingType.FOLDER_LOCATIONS,
+      SelectionComponentFactory.folderPath(ClientSetting.SAVE_GAMES_FOLDER_PATH),
+      "The folder where saved game files will be stored by default"),
+  USER_MAPS_FOLDER_PATH_BINDING(
+      "Maps Folder",
+      SettingType.FOLDER_LOCATIONS,
+      SelectionComponentFactory.folderPath(ClientSetting.USER_MAPS_FOLDER_PATH),
+      "The folder where game engine will download and find map files."),
+  WHEEL_SCROLL_AMOUNT_BINDING(
+      "Mouse Wheel Scroll Speed",
+      SettingType.MAP_SCROLLING,
+      SelectionComponentFactory.intValueRange(ClientSetting.WHEEL_SCROLL_AMOUNT, 10, 300),
+      "How fast the map will scroll when using the mouse wheel"),
+  PROXY_CHOICE(
+      "Network Proxy",
+      SettingType.NETWORK_PROXY,
+      SelectionComponentFactory.proxySettings(),
+      "Configure TripleA's Network and Proxy Settings\n"
+      + "This only effects Play-By-Forum games, dice servers, and map downloads.");
+
+
+  final SettingType type;
+  final String title;
+  final String description;
+  private final Supplier<SelectionComponent> selectionComponentBuilder;
+
+  private SelectionComponent selectionComponent;
+
+  ClientSettingUiBinding(
+      final String title,
+      final SettingType type,
+      final Supplier<SelectionComponent> selectionComponent,
```

Comment:
Should this parameter be named `selectionComponentBuilder`?  And should it have `checkNotNull()` applied?